views/wpmm-page-template.php

```diff
@@ -3,7 +3,18 @@
  * Template Name: Full Width for Maintenance Page
  */
 
-?><!DOCTYPE html>
+defined( 'ABSPATH' ) || exit;
+
+$settings              = WP_Maintenance_Mode::get_instance()->get_plugin_settings();
+$overrideable_template = wpmm_get_template_path( 'maintenance.php', true );
+
+if ( ( isset( $settings['design']['page_id'] ) && get_option( 'wpmm_new_look' ) ) && WPMM_VIEWS_PATH . 'maintenance.php' !== $overrideable_template ) {
+	if ( ! is_front_page() ) {
+		wp_redirect( home_url() );
+	}
+} else {
```

Comment:
The first two conditions are already checked before entering this file, so there is no need to check them again.

The overriding template is loaded only when the user gets to see the maintenance state of the site, so we could define a constant in `maintenance.php` like this `define( IS_MAINTENANCE, true )` to know when that happens. 

Otherwise, the user will just get redirected to the home page (which is not desired when the user is logged in)

```suggestion
$overrideable_template = wpmm_get_template_path( 'maintenance.php', true );
if ( defined( IS_MAINTENANCE ) && WPMM_VIEWS_PATH . 'maintenance.php' !== $overrideable_template ) {
	exit();
}

$settings = WP_Maintenance_Mode::get_instance()->get_plugin_settings();
```

Reply:
@arinaturcu, I've applied your suggestions. please review and do let me know if you have any queries.

Reply:
My bad here, we should add the `define( 'IS_MAINTENANCE', true )` in `includes/classes/wp-maintenance-mode.php` at line 688 after the checks to make it work:

https://github.com/Codeinwp/wp-maintenance-mode/blob/110a1dec6e52949d6f88fa97387b63e86bbdfde0/includes/classes/wp-maintenance-mode.php#L673-L687

Other than that, it's looking good, thanks! 🚀